interceptor/tracing_interceptor.go

```diff
@@ -886,14 +1010,38 @@ func (t *tracingInterceptor) writeSpanToHeader(span TracerSpan, header map[strin
 	return nil
 }
 
-func newErrFut(ctx workflow.Context, err error) workflow.ChildWorkflowFuture {
+func (t *tracingInterceptor) writeSpanToNexusHeader(span TracerSpan, header nexus.Header) error {
+	// Serialize span to map
+	data, err := t.tracer.MarshalSpan(span)
+	if err != nil || len(data) == 0 {
+		return err
+	}
+	// Put on header
+	for k, v := range data {
+		header.Set(k, v)
+	}
+	return nil
+}
+
+func (t *tracingInterceptor) readSpanFromNexusHeader(header nexus.Header) (TracerSpanRef, error) {
+	return t.tracer.UnmarshalSpan(header)
+}
+
+type anyWorkflowFuture interface {
```

Comment:
I wouldn't do this approach, I would create a different type to wrap the `NexusOperationFuture` and the `ChildWorkflowFuture`

Reply:
Do you think this matters enough to block the PR? Seems like the approaches are equivalent and the one I took doesn't require changing any existing code.

Reply:
I don't think It is correct to return a type implementing multiple different future interfaces from here , is is visible to users using reflection

Reply:
I can refactor if you think this is critical.